firmware/examples/FreeRTOS_Blinker/source/main.cpp

```diff
@@ -0,0 +1,83 @@
+#include <cstdio>
+#include <cstdint>
+
+#include "config.hpp"
+#include "FreeRTOS.h"
+#include "L0_LowLevel/LPC40xx.h"
+#include "L2_Utilities/debug_print.hpp"
+#include "task.h"
+
+namespace
+{
+void vLedToggle(void * pvParameters)
+{
+    SJ2_USED(pvParameters);
+    DEBUG_PRINT("Initializing LEDs in vLedToggle...");
+    LPC_IOCON->P1_1 &= ~(0b111);
+    LPC_IOCON->P1_8 &= ~(0b111);
+    LPC_GPIO1->DIR |= (1 << 1);
+    LPC_GPIO1->PIN &= ~(1 << 1);
+    LPC_GPIO1->DIR |= (1 << 8);
+    LPC_GPIO1->PIN |= (1 << 8);
+    DEBUG_PRINT("LEDs Initialized in vLedToggle...");
+    DEBUG_PRINT("Toggling LEDs in vLedToggle...");
+    while (true)
+    {
+        LPC_GPIO1->PIN ^= 0b0001'0000'0010;
+        vTaskDelay(100);
+    }
+}
+
+constexpr uint32_t kButtonPinNumber = 14;
+constexpr uint32_t kLedNumber       = 15;
+
+bool CheckSwitch3()
+{
+    return (LPC_GPIO1->PIN & (1 << kButtonPinNumber));
+}
+
+void vButtonReader(void * pvParameters)
+{
+    SJ2_USED(pvParameters);
+    DEBUG_PRINT("Initializing LED3 and SW3 in vButtonReader...");
+    DEBUG_PRINT("LEDs Initialized in vLedToggle...");
+    LPC_IOCON->P1_14 &= ~(0b111);
+    LPC_IOCON->P1_15 &= ~(0b111);
+    LPC_GPIO1->DIR &= ~(1 << kButtonPinNumber);
+    LPC_GPIO1->DIR |= (1 << kLedNumber);
+    LPC_GPIO1->PIN |= (1 << kLedNumber);
+    DEBUG_PRINT("LED3 and SW3 Initialized...");
+    DEBUG_PRINT("Press and release SW3 to toggle LED3 state...");
+    bool button_pressed = false;
+    while (true)
+    {
+        if (CheckSwitch3())
+        {
+            button_pressed = true;
+        }
+        else if (!CheckSwitch3() && button_pressed)
+        {
+            LPC_GPIO1->PIN ^= (1 << kLedNumber);
+            button_pressed = false;
+        }
+        else
+        {
+            button_pressed = false;
+        }
+        vTaskDelay(50);
+    }
+}
+
+}  // namespace
+
+int main(void)
+{
+    DEBUG_PRINT("Creating Tasks ...");
+    xTaskCreate(vLedToggle, "vLedToggle", 256, NULL, tskIDLE_PRIORITY + 1,
```

Comment:
Create const expressions for different priorities instrad of using tskIDLE_PRIORITY+1/+3/+etc..